webapp/lib/router.dart

```diff
@@ -1,28 +1,46 @@
 import 'dart:html';
+import 'controller.dart' as controller;
+
+class Route {
+  String path;
+  Function handler;
+
+  Route(this.path, this.handler);
+}
 
 class Router {
-  Map<String, Function> _routes;
+  Map<String, Route> _routes;
+  Route _authRoute;
 
   Router() {
     _routes = {};
   }
 
-  void addHandler(String route, Function callback) {
-    _routes[route] = callback;
+  void addHandler(Route route, {bool isAuthRoute = false}) {
```

Comment:
Let's have a separate method for the auth page instead of passing this boolean, e.g. `addAuthHandler`

Reply:
Added `addAuthHandler`